objects/src/builder/mod.rs

```diff
@@ -0,0 +1,266 @@
+use crate::{
+    mock::assembler, Account, AccountCode, AccountError, AccountId, AccountStorage, AccountType,
+    AccountVault, Asset, AssetError, Digest, Felt, FungibleAsset, MerkleStore, NonFungibleAsset,
+    NonFungibleAssetDetails, StorageItem, String, ToString, Vec, ZERO,
+};
+use assembly::ast::ModuleAst;
+use rand::{distributions::Standard, Rng};
+
+pub struct AccountStorageBuider {
+    items: Vec<StorageItem>,
+}
+
+/// Builder for an `AccountStorage`, the builder can be configured and used multipled times.
+impl AccountStorageBuider {
+    pub fn new() -> Self {
+        Self { items: vec![] }
+    }
+
+    pub fn add_item(&mut self, item: StorageItem) -> &mut Self {
+        self.items.push(item);
+        self
+    }
+
+    pub fn build(&self) -> AccountStorage {
+        // The account store treats duplicated keys as an error, but for convinience the builder
+        // allows it, the last entry is kept.
```

Comment:
what's the rationale for the builder allowing it?

Reply:
just a small convenience, would you prefer the other way around?

Reply:
sure, no strong feeling about this. removed the deduplication.